File: app/src/main/java/com/hqq/coreapp/adapter/MainAdapter.java
package com.hqq.coreapp.adapter;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.hqq.coreapp.R;
import com.hqq.coreapp.bean.MainBean;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.iblibrary.adapter
 * @FileName :   MainAdapter
 * @Date : 2018/11/22 0022  下午 7:45
 * @Descrive :
 * @Email :  qiqiang213@gmail.com
 */
public class MainAdapter extends BaseQuickAdapter<MainBean, BaseViewHolder> {
    public MainAdapter() {
        super(R.layout.item_main);
    }

    @Override
    public boolean isLoading() {
        return false;
    }

    @Override
    protected void convert(BaseViewHolder helper, MainBean item) {
        helper.setText(R.id.tv_title, item.getTitle());
    }
}

File: app/src/main/java/com/easy/example/adapter/MainAdapter.kt
package com.easy.example.adapter

import android.content.Context
import android.content.Intent
import android.view.View
import android.view.ViewGroup
import com.chad.library.adapter4.BaseQuickAdapter
import com.chad.library.adapter4.viewholder.QuickViewHolder
import com.easy.example.R
import com.easy.example.bean.MainBean

/**
 * @Author : huangqiqiang
 * @Package : com.easy.iblibrary.adapter
 * @FileName :   MainAdapter
 * @Date : 2018/11/22 0022  下午 7:45
 * @Describe :
 * @Email :  qiqiang213@gmail.com
 */
class MainAdapter : BaseQuickAdapter<MainBean<*>, QuickViewHolder>(),
    BaseQuickAdapter.OnItemClickListener<MainBean<*>> {

    init {

        setOnItemClickListener(this)

    }

    override fun onCreateViewHolder(context: Context, parent: ViewGroup, viewType: Int): QuickViewHolder {

        return QuickViewHolder(R.layout.item_main,parent)
    }

    override fun onBindViewHolder(helper: QuickViewHolder, position: Int, item: MainBean<*>?) {
        helper.setText(R.id.tv_title, item?.title)
    }

    override fun onClick(adapter: BaseQuickAdapter<MainBean<*>, *>, view: View, position: Int) {
        context.startActivity(Intent(context, getItem(position)?.className))
    }



}